### ISSUE TITLE
Don't warn about deprecated fields in implicitly generated special member functions

### ISSUE THREAD

**Issue by HolyBlackCat** [llvm/llvm-project#147293]
This gives a warning:
```cpp
struct A
{
    [[deprecated]] int x = 42;
};

int main()
{
    A a;
    A b = a;
    a = b;
}
```
```console
<source>:1:8: warning: 'A::x' is deprecated [-Wdeprecated-declarations]
    1 | struct A
      |        ^
<source>:3:24: note: declared here
    3 |     [[deprecated]] int x = 42;
      |                        ^
<source>:1:8: warning: 'A::x' is deprecated [-Wdeprecated-declarations]
    1 | struct A
      |        ^
<source>:3:24: note: declared here
    3 |     [[deprecated]] int x = 42;
      |                        ^
<source>: In function 'int main()':
<source>:8:7: note: synthesized method 'constexpr A::A()' first required here
    8 |     A a;
      |       ^
```
IMO this isn't very useful. Only the manual uses of this field should warn, not the implicit ones.

GCC has the same issue, but MSVC doesn't. I've tested on Clang 20 and trunk.

**Comment by shashforge** [comment]
Hello @HolyBlackCat/everyone,

Thank you for raising this issue—it looks like a valuable improvement to Clang’s diagnostics. I’d be honored to contribute my first PR by implementing a patch that suppresses deprecation warnings in implicit special‐member functions (default/copy constructors and copy‐assignment operators) via an early bail‐out in `Sema::DiagnoseUseOfDecl`. I can also add appropriate regression tests under `test/SemaCXX/`.

If it’s alright, could someone kindly assign this issue to me? I look forward to collaborating on this!

Thank you for your consideration. 😊
